tests/test_swift.py

```diff
@@ -69,44 +69,64 @@ def setUp(self):
         self.channels = set()
         for appInfo in APP_INFOS.values():
             self.channels.update(appInfo.channel)
-        self.channels = sorted(self.channels)
         self.swift = swift.Swift(APP_INFOS)
 
     def test_init(self):
         self.assertEqual(self.swift.appInfos, APP_INFOS)
-        self.assertIsInstance(self.swift.mainWindow, QMainWindow)
-        self.assertIsInstance(self.swift.centralWidget, QLabel)
         for name in APP_INFOS:
             self.assertIn(name, self.swift._dockWidgets)
             self.assertIn(name, self.swift._apps)
         for channel in self.channels:
             self.assertIn(channel, self.swift._subscribers)
 
     def test_app_names(self):
-        appNames = self.swift.appNames()
-        for name in APP_INFOS:
-            self.assertIn(name, appNames)
+        appNamesSet = set(self.swift.appNames())
+        self.assertEqual(appNamesSet, set(APP_INFOS))
+
+    def test_create_app(self):
+        app_ = MagicMock()
+        app_.frames.return_value = (QWidget(),)
+        cls = MagicMock(return_value=app_)
+        setattr(importlib.import_module.return_value, "cls3", cls)
+        self.swift.createApp(
+            "app3",
+            swift.AppInfo(**{"module": "module3", "cls": "cls3", "channel": ["ch1"]})
+        )
+        self.assertIn("app3", self.swift._apps)
+        self.assertIn("app3", self.swift._dockWidgets)
+        self.assertIn("app3", self.swift._subscribers["ch1"])
```

Comment:
Can you check if the `self.swift._apps["app3"]` is an instance of `cls` (i.e., "cls3")?

Reply:
I think it is the best way to set a name attribute of `app_` as "cls3" and check it.